### PR TITLE
Add manager tooltip disable and delay

### PR DESCRIPTION
**Pull request by CrazyBoy49z**
### What does it do?
Tooltip disable and delay

### Why is it needed?
This add tooltip disable and delay in manager

### Related issue(s)/PR(s)
#12732

### CODE REVIEW

**Review by Jako — DISMISSED** [review]

**Review by Jako — COMMENTED** [review]
Maybe the setting text could be changed, but that’s all. Thanks.

File: manager/assets/modext/util/utilities.js [hunk 1]
```diff
@@ -347,18 +347,20 @@ Ext.override(Ext.form.Action.Submit,{
 
 /* QTips to form fields */
 Ext.form.Field.prototype.afterRender = Ext.form.Field.prototype.afterRender.createSequence(function() {
-    if (this.description) {
+    if (this.description && (MODx.config.manager_tooltip_enable === true)) {
```
Comment (OptimusCrime) [inline manager/assets/modext/util/utilities.js:350]:
```
if (this.description && MODx.config.manager_tooltip_enable) {
```

?

Comment (CrazyBoy49z) [reply manager/assets/modext/util/utilities.js:350]: Test. It's not work

File: core/lexicon/en/setting.inc.php [hunk 2]
```diff
@@ -438,6 +438,9 @@
 $_lang['setting_manager_login_url_alternate'] = 'Alternate Manager Login URL';
 $_lang['setting_manager_login_url_alternate_desc'] = 'An alternate URL to send an unauthenticated user to when they need to login to the manager. The login form there must login the user to the "mgr" context to work.';
 
+$_lang['setting_manager_tooltip_enable'] = 'Manager tooltip enable';
```
Comment (JoshuaLuckers) [inline core/lexicon/en/setting.inc.php]:
```suggestion
$_lang['setting_manager_tooltip_enable'] = 'Enable Manager Tooltips';
```
Please consider adding a descriptive lexicon `setting_manager_tooltip_enable_desc` explaining the setting.

File: core/lexicon/en/setting.inc.php [hunk 3]
```diff
@@ -438,6 +438,9 @@
...
+$_lang['setting_manager_tooltip_delay'] = 'Manager tooltip delay';
```
Comment (JoshuaLuckers) [inline core/lexicon/en/setting.inc.php]:
```suggestion
$_lang['setting_manager_tooltip_delay'] = 'Delay Time for Manager Tooltips';
```

Please consider adding a descriptive lexicon `setting_manager_tooltip_delay ` explaining the setting and value.

File: manager/assets/modext/util/utilities.js (hunk 1 above)
Comment (JoshuaLuckers) [inline manager/assets/modext/util/utilities.js:350]:
```suggestion
    if (this.description && MODx.config.manager_tooltip_enable) {
```

**Review by JoshuaLuckers — CHANGES_REQUESTED** [review]

**Comment by CrazyBoy49z** [comment]
if (this.description && MODx.config.manager_tooltip_enable) {

this fixes not works.

**Comment by JoshuaLuckers** [comment]
> if (this.description && MODx.config.manager_tooltip_enable) {
> 
> this fixes not works.

It works fine for me. Make sure you have the system setting `compress_js` set to `false`.

**Review by CrazyBoy49z — COMMENTED** [review]
> > if (this.description && MODx.config.manager_tooltip_enable) {
> > this fixes not works.
> 
> It works fine for me. Make sure you have the system setting `compress_js` set to `false`.

Thank you but I'm checking with compress_js = false. 

manager_tooltip_enable = false

When "if (this.description && (MODx.config.manager_tooltip_enable===true))" not show tooltips. 

When "if (this.description && MODx.config.manager_tooltip_enable)" show tooltips

**Comment by JoshuaLuckers** [comment]
> > > if (this.description && MODx.config.manager_tooltip_enable) {
> > > this fixes not works.
> > 
> > 
> > It works fine for me. Make sure you have the system setting `compress_js` set to `false`.
> 
> Thank you but I'm checking with compress_js = false.
> 
> manager_tooltip_enable = false
> 
> When "if (this.description && (MODx.config.manager_tooltip_enable===true))" not show tooltips.
> 
> When "if (this.description && MODx.config.manager_tooltip_enable)" show tooltips

What's the value when you open the developer console of your browser and insert `MODx.config.manager_tooltip_enable`?

**Comment by CrazyBoy49z** [comment]
> > > > if (this.description && MODx.config.manager_tooltip_enable) {
> > > > this fixes not works.
> > > 
> > > 
> > > It works fine for me. Make sure you have the system setting `compress_js` set to `false`.
> > 
> > 
> > Thank you but I'm checking with compress_js = false.
> > manager_tooltip_enable = false
> > When "if (this.description && (MODx.config.manager_tooltip_enable===true))" not show tooltips.
> > When "if (this.description && MODx.config.manager_tooltip_enable)" show tooltips
> 
> What's the value when you open the developer console of your browser and insert `MODx.config.manager_tooltip_enable`?

return "0"

image

**Review by JoshuaLuckers — APPROVED** [review]

**Comment by OptimusCrime** [comment]
@CrazyBoy49z Ah, okey, makes sense. Keep it like it was, it is fine.

File: manager/assets/modext/util/utilities.js (hunk 1 above)
Comment (Mark-H) [reply manager/assets/modext/util/utilities.js:350]: I'm like 95% sure the check as it is now (`=== true`) is not going to work. The config object is not strictly typed, as settings are stored as text, so a strict type check for true will always be false. The value of `MODx.config.manager_tooltip_enable` will in most cases be a string `'1'` or `'0'`/`''`.

**Review by Mark-H — APPROVED** [review]

**Comment by Mark-H** [comment]
[As mentioned before](https://github.com/modxcms/revolution/pull/14535#discussion_r278476532), the strict type check does not work. I've tweaked the check a little to make it work and merged it into 3.x. Thanks.

**Pull request merged**